sources/Windowing/Windowing/Implementations/SDL3/SdlSurfaceComponents.OpenGL.cs

```diff
@@ -0,0 +1,431 @@
+            SetEnabled();
+            field = value;
+        }
+    }
+
+    public int? PreferredStencilBufferBits
+    {
+        get;
+        set
+        {
+            AssertUninit();
+            SetEnabled();
+            field = value;
+        }
+    }
+
+    public Vector4D<int>? PreferredBitDepth
+    {
+        get;
+        set
+        {
+            AssertUninit();
+            SetEnabled();
+            field = value;
+        }
+    }
+
+    public int? PreferredSampleCount
+    {
+        get;
+        set
+        {
+            AssertUninit();
+            SetEnabled();
+            field = value;
+        }
+    }
+
+    public Version32? Version
+    {
+        get;
+        set
+        {
+            AssertUninit();
+            SetEnabled();
+            field = value;
+        }
+    }
+
+    public OpenGLContextFlags Flags
+    {
+        get;
+        set
+        {
+            AssertUninit();
+            SetEnabled();
+            field = value;
+        }
+    }
+
+    public OpenGLContextProfile Profile
+    {
+        get;
+        set
+        {
+            AssertUninit();
+            field = value;
+        }
+    }
+
+    public bool IsSupported => TryCreateContext(false);
+
+    public bool ShouldSwapAutomatically { get; set; } = true;
+
+    public IGLContext? SharedContext
+    {
+        get;
+        set
+        {
+            AssertUninit();
+            SetEnabled();
+            field = value;
+        }
+    }
+
+    private void SetEnabled()
+    {
+        if (Profile == OpenGLContextProfile.None)
+        {
+            Profile = OpenGLContextProfile.Default;
+        }
+    }
+
+    private bool TryCreateContext(bool isInitializing)
+    {
+        var profile = Profile;
+        if (profile is OpenGLContextProfile.None)
+        {
+            return true;
+        }
+
+        // If isInitializing == false, we should restore the thread to the state we found it in.
+        var current = Sdl.GLGetCurrentContext();
+        WindowHandle currentWindow = nullptr;
+        if (current != nullptr)
+        {
+            currentWindow = Sdl.GLGetCurrentWindow();
+        }
+
+        if (Handle == nullptr)
+        {
+            if (isInitializing)
+            {
+                throw new InvalidOperationException(
+                    "Attempted to initialize OpenGL context before window initialization - this should not be possible "
+                        + "with normal usage. Please report this at https://github.com/dotnet/Silk.NET"
+                );
+            }
+
+            // We need to create a dummy window if we haven't already (InitializePlatform does this, but it's only made
+            // available to the first Surface).
+            var props = CreateDummyWindowProps();
+            Handle = Sdl.CreateWindowWithProperties(props);
+            Sdl.DestroyProperties(props);
+            if (Handle == nullptr)
+            {
+                Sdl.ThrowError();
```

Comment:
This needs some thought

Reply:
This is now just a `return false` as a dummy window failing to be created with the OpenGL flags indicates non-support.